lib/presentation/auth/auth_screen.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:dots_indicator/dots_indicator.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../application/auth/auth_bloc.dart';
+import '../routes/app_routes.gr.dart';
+import '../shared_widgets/custom_app_bars/custom_appbar.dart';
+import '../themes/constants.dart';
+import '../utils/context.ext.dart';
+import 'pages/enter_username.dart';
+import 'pages/profile_photo.dart';
+import 'pages/verification_code.dart';
+import 'pages/verify_phone.dart';
+
+class AuthPage extends StatefulWidget {
+  const AuthPage({Key? key}) : super(key: key);
+
+  @override
+  _AuthPageState createState() => _AuthPageState();
+}
+
+class _AuthPageState extends State<AuthPage> {
+  // All Pages
+  final _pageController = PageController();
+  double _currentPage = 0.0;
+  late List<Widget> _pages;
+
+  bool _displayDots = true;
+
+  // Page One
+  // final GlobalKey<VerifyPhonePageState> _verifyPhoneKey = GlobalKey();
+
+  // Page Two
+
+  @override
+  void initState() {
+    super.initState();
+    // All pages
+    _pages = [
+      const VerifyPhonePage(/*key: _verifyPhoneKey*/),
+      const EnterVerificationCode(),
+      const EnterUserName(),
+      SelectProfilePhoto(
+        onSkip: () => _authDone(context),
+      )
+    ];
+
+    _pageController.addListener(() {
+      setState(() {
+        _currentPage = _pageController.page!;
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => getIt<AuthBloc>(),
+      child: BlocListener<AuthBloc, AuthState>(
+        listener: (context, state) {
+          state.maybeMap(
+              smsCodeSent: (_) => _toPage(1),
+              loggedIn: (loggedInState) {
+                if (loggedInState.isNewUser) {
+                  // TODO - Handle if is existing user
+                } else {
+                  _toPage(2);
+                }
+              },
+              verificationCompleted: (_) {},
+              authError: (authError) {
+                context.showErrorSnack(
+                  authError.failure.map(
+                    serverError: (_) => "Server Error",
+                    invalidPhone: (_) => "Invalid Phone",
+                    verificationFailed: (_) => "Verification Failed",
+                    networkRequestFailed: (_) => "No Internet connection",
+                    invalidSmsCode: (_) => "Invalid SMS Code",
+                  ),
+                );
+              },
+              usernameUpdateDone: (_) {
+                _toPage(3);
+                setState(() => _displayDots = false);
+              },
+              photoUpdateDone: (_) {
+                _authDone(context);
+              },
+              orElse: () {});
+        },
+        child: Scaffold(
+          resizeToAvoidBottomInset: true,
+          appBar: _currentPage == 0
+              ? CustomAppBar(
```

Comment:
We don't have CustomAppBar anymore, need to rebase on the remote development branch and use CleanAppBar.